hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -1090,7 +1090,7 @@ DatanodeInfo[] getCachedDatanodeReport(DatanodeReportType type)
       throws IOException {
     try {
       DatanodeInfo[] dns = this.dnCache.get(type);
-      if (dns == null) {
+      if (dns.length == 0) {
```

Comment:
I think it is safer to check for both null and length.

Reply:
done.